src/shared/vpick.c

```diff
@@ -348,7 +352,7 @@ static int make_choice(
 
                         e++;
                 } else
-                        e = (*entry)->d_name;
+                        e = dname;
 
                 if (!isempty(filter->suffix)) {
```

Comment:
Maybe do this in the else branch of this if statement? Or, `chopped` can be dropped can we can chop `e` directly.